### PR TITLE
allow for direct classes

### PR DESCRIPTION
**Pull request by h4ck3rm1k3**
This is more of a hack but would like top open a discussion.

The patch will allow you to specify a class directly instead of a path name for loading. 

Usage here  https://github.com/flosokaks/scrapy_ks/blob/master/scrapy_openkansas.py

### CODE REVIEW

**Comment by kmike** [comment]
I like this feature.

File: scrapy/utils/misc.py [hunk 1]
```diff
@@ -35,13 +35,15 @@ def load_object(path):
...
-
-    module, name = path[:dot], path[dot+1:]
-    mod = import_module(module)
+    if isinstance(path,str) :
+        if '.' in path :
+            dot = path.rindex('.')
+            module, name = path[:dot], path[dot+1:]
+            mod = import_module(module)
+        else:
+            raise Exception()
```
Comment (kmike) [inline scrapy/utils/misc.py]: this looks rather uninformative - when is it raised?

Comment (h4ck3rm1k3) [inline scrapy/utils/misc.py]: yes the exception should be 
   raise ValueError("Error loading object '%s': not a full path" % path)
 not raise exception.

Here is how I am using it :
https://github.com/flosokaks/scrapy_ks/blob/master/scrapy_openkansas.py#L293

settings.set('ITEM_PIPELINES',{
PythonCache : 300,
MarkDownCache : 300,
HtmlCache : 300,
})

**Comment by Gallaecio** [comment]
Done in #3873

**Pull request closed without merging**